tests/global/test_main.py

```diff
@@ -0,0 +1,80 @@
+import argparse
+import sys
+import unittest
+
+from ert_gui.main import ert_parser, main
+
+if sys.version_info >= (3, 3):
+    from unittest.mock import Mock, patch
+else:
+    from mock import Mock, patch
+
+
+class MainTest(unittest.TestCase):
+
+    # main is essentially a call to ArgumentParser.parse_args, so it is mocked.
+    arg_parser_mock = None
+
+    # ArgumentParser.parse_args returns a Namespace, which must be mocked as
+    # parse_args is stubbed out.
+    namespace_mock = None
+
+    def mock_parser(self):
+        """Returns a mocked ArgumentParser."""
+        return self.arg_parser_mock
+
+    @classmethod
+    def setUpClass(cls):
+        # argv is inconsequential for this case, as parsing is stubbed out.
+        # Tests still needs to assert argv was passed, and since it may vary
+        # (based on environment) it is unset.
+        sys.argv = list()
+
+        cls.namespace_mock = Mock(spec=argparse.Namespace)
```

Comment:
Any reason why we would not add these to `self` via `setUp`  instead and hence not have to deal with the `tearDown` at all?

Reply:
I think this thinking suited for Go-projects. I don' know.

Addressed!